src/libraries/System.Net.Security/src/System/Net/Security/MD4.cs

```diff
@@ -0,0 +1,245 @@
+            // Zeroize sensitive information
+            Array.Clear(buffer, 0, 64);
+            Array.Clear(x, 0, 16);
+
+            HashCore(source.ToArray(), 0, source.Length, state);
+            HashValue = HashFinal(state);
+            byte[] result = (byte[]) HashValue.Clone();
+            result.CopyTo(destination);
+        }
+
+        private static byte[]? HashValue;
+        private static byte[] buffer = new byte[64];
+        private static uint[] count = new uint[2];
+        private static uint[] x = new uint[16];
+
+        private const int S11 = 3;
+        private const int S12 = 7;
+        private const int S13 = 11;
+        private const int S14 = 19;
+        private const int S21 = 3;
+        private const int S22 = 5;
+        private const int S23 = 9;
+        private const int S24 = 13;
+        private const int S31 = 3;
+        private const int S32 = 9;
+        private const int S33 = 11;
+        private const int S34 = 15;
+
+        private static byte[] digest = new byte[16];
+        private static void HashCore(byte[] array, int ibStart, int cbSize, Span<uint> state)
+        {
+            /* Compute number of bytes mod 64 */
+            int index = (int)((count[0] >> 3) & 0x3F);
+            /* Update number of bits */
+            count[0] += (uint)(cbSize << 3);
+            if (count[0] < (cbSize << 3))
+                count[1]++;
+            count[1] += (uint)(cbSize >> 29);
+
+            int partLen = 64 - index;
+            int i = 0;
+            /* Transform as many times as possible. */
+            if (cbSize >= partLen)
+            {
+                Buffer.BlockCopy(array, ibStart, buffer, index, partLen);
+                MD4Transform(state, buffer, 0);
+
+                for (i = partLen; i + 63 < cbSize; i += 64)
+                {
+                    MD4Transform(state, array, ibStart + i);
+                }
+
+                index = 0;
+            }
+
+            /* Buffer remaining input */
+            Buffer.BlockCopy(array, ibStart + i, buffer, index, (cbSize - i));
+        }
+
+        private static byte[] HashFinal(Span<uint> state)
+        {
+            /* Save number of bits */
+            byte[] bits = new byte[8];
+            Encode(bits, count);
+
+            /* Pad out to 56 mod 64. */
+            uint index = ((count[0] >> 3) & 0x3f);
+            int padLen = (int)((index < 56) ? (56 - index) : (120 - index));
+            HashCore(Padding(padLen), 0, padLen, state);
+
+            /* Append length (before padding) */
+            HashCore(bits, 0, 8, state);
+
+            /* Store state in digest */
+            Encode(digest, state);
+
+            return digest;
+        }
+
+        //--- private methods ---------------------------------------------------
+
+        private static byte[] Padding(int nLength)
+        {
+            byte[] padding = new byte[nLength];
+            padding[0] = 0x80;
+            return padding;
+        }
```

Comment:
Instead of allocating an array, the call site should just have this as `Span<byte> padding = stackalloc byte[padLen]; padding[0] = 0x80;`